src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Sentry.Internal.Extensions
+{
+    internal static class JsonExtensions
+    {
+        public static void WriteDictionary(
+            this Utf8JsonWriter writer,
+            string propertyName,
+            IReadOnlyDictionary<string, string?>? dic)
+        {
+            writer.WritePropertyName(propertyName);
```

Comment:
I guess avoiding the while object (not writing prop name) if there are not values is the behavior we want (so far hacked with the null backing fields, which for bonus avoideed allocations that weren't needed/lazy init)